ipc/provider/src/manager/evm/manager.rs

```diff
@@ -100,32 +100,65 @@ impl TopDownCheckpointQuery for EthSubnetManager {
     async fn get_top_down_msgs(
         &self,
         subnet_id: &SubnetID,
-        start_epoch: ChainEpoch,
-        end_epoch: ChainEpoch,
-    ) -> Result<Vec<ipc_sdk::cross::CrossMsg>> {
-        self.top_down_msgs(subnet_id, start_epoch, end_epoch).await
+        epoch: ChainEpoch,
+    ) -> Result<TopDownQueryPayload<Vec<CrossMsg>>> {
+        let route = subnet_id_to_evm_addresses(subnet_id)?;
+        log::debug!("getting top down messages for route: {route:?}");
+
+        let subnet_id = gateway_getter_facet::SubnetID {
+            root: subnet_id.root_id(),
+            route,
+        };
+        let gateway_contract = gateway_getter_facet::GatewayGetterFacet::new(
+            self.ipc_contract_info.gateway_addr,
+            Arc::new(self.ipc_contract_info.provider.clone()),
+        );
+        let (raw_msgs, hash) = gateway_contract
+            .method::<_, (Vec<gateway_getter_facet::CrossMsg>, [u8; 32])>(
+                "getTopDownMsgs",
+                gateway_getter_facet::GetTopDownMsgsCall {
+                    subnet_id,
+                    block_number: U256::from(epoch),
+                },
+            )
+            .map_err(|e| anyhow!("cannot create the top down msg call: {e:}"))?
+            .call()
```

Comment:
For reference why I don't think this won't work: https://github.com/consensus-shipyard/ipc-solidity-actors/pull/240/files#r1358015064

That said, there is an option to set the `BlockId` of a `ContractCall` before it's sent, but I have no idea what would happen in the EVM:
* currently we send a call to `getTopDownMsgs(height)` at the latest block height, so if `height < latest.height - 256` this will not work
* we could _also_ set the `height` on the `ContractCall` itself, so that _relatively_ it wouldn't be trailing the queried block by 256, it would match, but the EVM might not care

Reply:
The EVM actor compares always to the [curr_epoch](https://github.com/filecoin-project/builtin-actors/blob/master/actors/evm/src/interpreter/instructions/context.rs#L25) of the runtime. That might be initialized to where the query was run, at least that's what I do in Fendermint, not sure about Lotus.